src/transformers/modeling_utils.py

```diff
@@ -846,9 +846,7 @@ def generate(
             encoder_inputs = input_ids
             input_ids = torch.full(
                 (effective_batch_size * num_beams, 1),
-                eos_token_id,
-#                bos_token_id,
-                #                eos_token_id, # Why eos_token_id here? bos_token_id seems to work as well ... to see if it works as well with hard summarization case
+                eos_token_id,  # TODO (PVP): to check if this is the only solution -> quite hacky to do this
```

Comment:
fairseq way is even crazier! look for `prefix_tokens` in sequence_generator.py

Reply:
Ok, not sure whether all `encoder-decoder` models should start with `eos_token_id` though. Happy to discuss!

Reply:
(read the prefix tokens comment before this one)
My understanding is that since we know we are going to generate BOS first, we want the score associated with BOS to be P(BOS | EOS), because that seems reasonable. So the first step, even though we will force BOS, we send EOS as input_ids to get the appropriate score.